libc-top-half/musl/src/thread/thrd_sleep.c

```diff
@@ -1,11 +1,17 @@
 #include <threads.h>
 #include <time.h>
 #include <errno.h>
+#ifdef __wasilibc_unmodified_upstream
 #include "syscall.h"
+#endif
 
 int thrd_sleep(const struct timespec *req, struct timespec *rem)
 {
+#ifdef __wasilibc_unmodified_upstream
 	int ret = -__clock_nanosleep(CLOCK_REALTIME, 0, req, rem);
+#else
+	int ret = -clock_nanosleep(CLOCK_REALTIME, 0, req, rem);
+#endif
```

Comment:
Add `weak_alias(__clock_nanosleep, clock_nanosleep);` to the file where clock_nanosleep is defined (like musl does) ... to avoid this diff?

Reply:
Hm, not sure I understand the "why" of what you're proposing: then we would have a weak alias in one direction in the bottom half and another weak alias in the other direction in the top half? (See `libc-top-half/musl/src/time/clock_nanosleep.c`).

Reply:
We don't compile the top half version of clock_nanosleep right?  IIUC it gets completely replaced by the bottom half version in `libc-bottom-half/cloudlibc/src/libc/time/clock_nanosleep.c`, right?

I'm suggesting that you make the bottom half version match the top half version by adding `weak_alias(__clock_nanosleep, clock_nanosleep);` at the end the file.. then this patch would not need needed.

Reply:
Ok, didn't understand the "non-compiled top-half bits" of this comment; thanks. I ad